File: User_management/backend/updateUser/dbconnection.py
import pymysql
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv(dotenv_path='../../.env')

# Database configuration
DB_CONFIG = {
    "host": os.getenv("DB_HOST"),
    "user": os.getenv("DB_USER"),
    "password": os.getenv("DB_PASSWORD"),
    "database": os.getenv("DB_NAME"),
    "port": int(os.getenv("DB_PORT"))
}

# Function to get a new database connection
def get_db_connection():
    try:
        return pymysql.connect(**DB_CONFIG)
    except pymysql.MySQLError as e:
        print(f"Error connecting to the database: {str(e)}")
        return None

File: User_management/backend/listUsers/dbConnection.rb
require 'mysql2'
require 'dotenv'

# Load environment variables from a specific .env file
Dotenv.load('../../.env')

def connect_to_db
  begin
    # Establish database connection using .env variables
    client = Mysql2::Client.new(
      host: ENV['DB_HOST'],
      username: ENV['DB_USER'],
      password: ENV['DB_PASSWORD'],
      database: ENV['DB_NAME'],
      port: 3306
    )
    puts "Connected to the database successfully"
    return client
  rescue Mysql2::Error => e
    # Handle connection error
    puts "Error connecting to the database: #{e.message}"
    return nil
  end
end

# Export global database connection
$db_client = connect_to_db
